### ISSUE TITLE
Doesn't try to find `.stylelintrc` file

### ISSUE THREAD

**Issue by anseki** [shinnn/vscode-stylelint#2]
Hi, thank you for your nice extension.

So, it seems that it has a problem.

`.stylelintrc` file in a working directory or ancestor directories never be used, because stylelint is passed a `config` object (even if it is empty) everytime.

https://github.com/stylelint/stylelint/blob/master/docs/user-guide/node-api.md#configoverrides

> If any config object is passed, stylelint does not bother looking for a .stylelintrc file

Therefore stylelint never try to find `.stylelintrc` file.
Has to add to VSCode settings to use `.stylelintrc` file in current working directory:

```json
"stylelint.config": { "extends": "./.stylelintrc" }
```

**Comment by benfrain** [comment]
I seem to have the same issue. However, having tried adding:

`"stylelint.config": { "extends": "./.stylelintrc" }`

As @anseki suggests I get this error:

`stylelint: Cannot read property 'split' of undefined`

**Comment by corysimmons** [comment]
Doesn't find stylelint.config.js either.

Why even put this text in your README if it's not true?

![image](https://cloud.githubusercontent.com/assets/224636/12972403/d77cbff8-d072-11e5-8060-9b5a625c6864.png)

**Comment by shinnn** [comment]
I just published a new version.

@anseki @benfrain Would you mind trying vscode-stylelint v0.3.0? At least this issue is fixed in my Mac.

**Comment by benfrain** [comment]
Hi, I Installed v0.3.0 but I'm afraid I see the same error

**Comment by shinnn** [comment]
@benfrain What do you mean by "the same error"? In my editor vscode-stylelint successfully loads the config file.

![image](https://cloud.githubusercontent.com/assets/1131567/12995314/ddfd8518-d169-11e5-9fd7-dd26311f765c.png)

Can you show me:

1. User and workspace settings (especially `stylelint.config`, `stylelint.configOverrides`)
2. Directory structure (the CSS file you tries to lint, and the config file you tries to load)

**Comment by anseki** [comment]
Hi, I tried v0.3.0, then it seems to be working fine.
The `.stylelintrc` file in both current working directory or ancestor directories was found, and stylelint changed its behavior by settings in these files.

**Comment by shinnn** [comment]
@anseki Thanks for the confirmation.

**Comment by benfrain** [comment]
Hi @shinnn  

When I open a CSS file I get the error `stylelint: Cannot read property 'split' of undefined`

My settings.json:

````
// Place your settings in this file to overwrite the default settings
{
	// Controls the font size.
	"editor.fontSize": 16,
    "stylelint.enable": true
}
````

My config is via a .stylelintrc file in the root folder. Stylelint behaves as expected using Sublime and https://github.com/kungfusheep/SublimeLinter-contrib-stylelint but I see the error above in vscode with this plugin. Should I be configuring differently?

**Comment by shinnn** [comment]
@benfrain Can you show your `.stylelintrc`?

**Comment by benfrain** [comment]
Hi, sure:

````
{
    "plugins": [
        "./.custom_linting_rules/css/onlyOverridesNested",
        "./.custom_linting_rules/css/mustMatchECSSNaming",
        "./.custom_linting_rules/css/keySelectorsMustNotBeCompound",
        "./.custom_linting_rules/css/keySelectorsMustBeAlone",
        "stylelint-statement-max-nesting-depth",
        "./.custom_linting_rules/css/componentNameMustMatchFileName"
    ],
    "rules": {
        "block-no-empty": true,
        "color-no-invalid-hex": true,
        "declaration-colon-space-after": "always",
        "declaration-colon-space-before": "never",
        "function-comma-space-after": "always",
        "function-url-quotes": "double",
        "media-feature-colon-space-after": "always",
        "media-feature-colon-space-before": "never",
        "media-feature-name-no-vendor-prefix": true,
        "max-empty-lines": 2,
        "number-leading-zero": "never",
        "number-no-trailing-zeros": true,
        "property-no-vendor-prefix": true,
        "rule-no-duplicate-properties": true,
        "declaration-block-no-single-line": true,
        "rule-trailing-semicolon": "always",
        "selector-list-comma-newline-after": "always-multi-line",
        "selector-no-id": true,
        "string-quotes": "double",
        "value-no-vendor-prefix": true,
        "function-linear-gradient-no-nonstandard-direction": true,
        "selector-no-universal": true,
        "rule-no-shorthand-property-overrides": true,
        "indentation": 4,
        "only-overrides-nested": true,
        "must-match-ecss-naming": true,
        "key-selectors-must-not-be-compound": true,
        "key-selectors-must-be-alone": true,
        "component-name-must-match-file-name": true,
        "statement-max-nesting-depth": [3, { countAtRules: false }]
    }
}
````

**Comment by shinnn** [comment]
@benfrain Thanks. It seems that plugin path resolution doesn't work correctly in the current version. I7ll look into this further.

**Comment by benfrain** [comment]
Just updated that prior message having attempted a reduction. Not sure if that adds anything relevant?

**Comment by shinnn** [comment]
@benfrain Can you remove [these lines](https://github.com/shinnn/vscode-stylelint/blob/984a53f0106d6e2cfca2fa71a8a83af40d231ec2/server.js#L39-L41) in your local vscode-stylelint? I hope this solves the `Cannot read property 'split'` error.

**Comment by benfrain** [comment]
Forgive my ignorance - what's the path to get to this file?

**Comment by shinnn** [comment]
@benfrain If you haven't changed the default editor settings, all extensions are placed in `~/.vscode/extensions/`.

**Comment by benfrain** [comment]
Removed those but same error: `stylelint: Cannot read property 'split' of undefined`

**Comment by shinnn** [comment]
Published v0.3.1. It doesn't include actual fix for this problem, but has an error message improvement.

@benfrain Can you try v0.3.1? Then we can see which line of code causes the `Cannot read property 'split'` error.

**Comment by shinnn** [comment]
Closing this, because the original issue "Doesn't try to find `.stylelintrc` file" is fixed now.

@benfrain Feel free to create a new issue for the `Cannot read property` problem.
